packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+
+## Current release process
+
+The previous is implemented (more or less accurate) during the release process. To start a new release a user needs to run `npm run release` that executes `scripts/release.ts`. The gist is:
+
+1. Calculate what packages need to be published. To achieve this, the script searches the latest tag for each package and checks if there have been any commits since then that affect it.
+1. Sort the dependencies by less dependants, i.e. if `packageA` depends on `packageB` it will work on `packageB` first.
+1. For each package:
+   * Bump the `version` in the `package.json` of the package using the related commit message(s) and update `CHANGELOG.md`
+   * Clean the package, intall dependencies via `npm`, and run `npm run test-release`
+   * Publish the new package to npm
+   * Update references from other packages
+   * Publish changes into GitHub
+
+This script also takes care of removing the `private` property from `package.json`, initializes a new `CHANGELOG.md` if needed, downloads files that are updated elsewhere like snyk's database, and such.
+
+## Current problems
+
+While the current process works most of the times OK, there are a few things that could be improved:
+
+* Using `tag`s to calculate the list of changes is not reliable. The team has experienced issues with tags being different accross forks, and such.
+* Installing each package individually is a slow process. There used to be problems with dependencies missing and such, that can probably be avoided with the right tooling.
+
+  In the past, we had problems with dependencies declared in one `package.json` being found in other packages. This happened due to workspaces hoisting the dependencies.
+* Another issue using npm to install the dependencies of each package individually is that the packages need to be published in the right order and npm needs to be updated so there isn't any problem requesting the dependency. There have been instances where npm's index wasn't yet updated and the process crashed because the wanted version wasn't available. Another big issue with this approach is that there is currently a circular dependency that if not handled correctly could cause an infinite publish loop.
+* Because each package is tested individually when doing the release (contrary to testing everything first before releasing) the release process might stop in the middle because one of the packages fails for whatever reason.
+* It's not clear how to prevent publishing a package, the script seems to remove the `private` property and I couldn't find any `ignorePackages` or similar in the code.
+* The script has never been run to its end successfully on Windows and testing changes is complicated.
+* The release can not be fully automated. It uses OTP for GitHub and npm and doesn't take into account other type of authentication mechanisms.
+* Other minor issues like can't continue even if tests fail, auto approve changelog, release an specific package, etc.
+
```

Comment:
Expand the `configuration` cases here to decide what to do if a dependency breaks, etc.

Reply:
@antross I've expanded this section with a table. I think it should cover all cases nicely and follow `semver` more strictly. Let me know if you find another case or disagree in any of these!